purchase_amendment/model/purchase_order_amendment.py

```diff
@@ -0,0 +1,278 @@
+                                           related='procurement_id.group_id',
+                                           readonly=True)
+    origin = fields.Char(readonly=True)
+    move_id = fields.Many2one(comodel_name='stock.move',
+                              readonly=True)
+    original_qty = fields.Float(string='Ordered',
+                                digits_compute=dp.get_precision('Product UoS'),
+                                readonly=True)
+    new_qty = fields.Float(string='Amend',
+                           digits_compute=dp.get_precision('Product UoS'))
+    state = fields.Char(readonly=True)
+    product_id = fields.Many2one(related='purchase_line_id.product_id',
+                                 readonly=True)
+    product_uom_id = fields.Many2one(related='purchase_line_id.product_uom',
+                                     readonly=True)
+
+    def affected_sale_lines(self):
+        move_chain = self.env['stock.move']
+        current_move = self.move_id
+        if current_move:
+            move_chain |= current_move
+            current_move = current_move.move_dest_id
+
+        return move_chain.mapped('procurement_id.sale_line_id')
+
+    @api.multi
+    def split_lines(self):
+        """ Split the order line according to selected quantities
+
+        The received quantity is the quantity that will remain in the original
+        purchases order line.
+        The canceled quantity will split the original line and cancel the
+        duplicated line.
+        The amended quantity will split the original line; the
+        duplicated line will be 'confirmed' and a new picking will be created.
+        """
+        Move = self.env['stock.move']
+        for item in self:
+            line = item.purchase_line_id
+            rounding = line.product_id.uom_id.rounding
+            compare = partial(float_compare, precision_digits=rounding)
+
+            if compare(item.original_qty, item.new_qty) == 0:  # no changes
+                continue
+            elif compare(item.new_qty, 0.) == 0:  # new quantity is zero
+                if compare(item.original_qty,
+                           line.product_qty) == 0:
+                    # this purchase line has only one move, and quantity goes
+                    # to zero: cancel the purchase order line
+                    line.action_cancel()
+                else:
+                    # we cancel the move, but the purchase order line has some
+                    # quantity left from other moves
+                    values = {'product_qty': item.move_id.product_qty,
+                              'amend_id': line.id,
+                              'move_ids': [(6, 0, item.move_id.ids)],
+                              'procurement_ids': [
+                                  (6, 0, item.move_id.procurement_id.ids)
+                              ]}
+                    cancel_line = line.copy(default=values)
```

Comment:
Add item.move_id.action_cancel() here (before cancel_line.action_cancel(). Otherwise the PO will not be cancelled when all lines are amended to 0